test/publisher/flow-control.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This will let us dig into the innards of the flow controller in peace.
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import * as assert from 'assert';
+import {describe, it} from 'mocha';
+
+import * as fc from '../../src/publisher/flow-control';
+
+describe('Flow Controller', () => {
+  const optionsDefault: fc.FlowControlOptions = {
+    maxOutstandingMessages: 5,
+    maxOutstandingBytes: 100,
+  };
+
+  it('does basic bookkeeping correctly', async () => {
+    const flowPause = new fc.FlowControl(optionsDefault);
+    await flowPause.willSend(10, 1);
+    assert.strictEqual((flowPause as any).bytes, 10);
```

Comment:
huh - why the `as any` casts?

Reply:
The `.bytes` is private on the class. This is poking in to check the counters. There's probably no terribly compelling reason not to expose this as a readonly getter, so I might do that.